hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -264,6 +268,31 @@ public AbfsConfiguration getAbfsConfiguration() {
     return this.abfsConfiguration;
   }
 
+  public void access(Path path, FsAction mode) throws IOException {
+    LOG.debug("access for filesystem: {}, path: {}, mode: {}",
+        this.client.getFileSystem(), path, mode);
+    if (!this.abfsConfiguration.isCheckAccessEnabled()) {
+      return;
+    }
+    String relativePath = path.isRoot() ? "" : ("/" + getRelativePath(path));
+    throwFileNotFoundIfNotExists(relativePath);
+    AbfsRestOperation abfsOp = this.client
+        .checkAccess(relativePath, mode.SYMBOL);
+    if (abfsOp.getResult().getStatusCode() != CHECK_ACCESS_SUCCESS) {
+      throw new AccessControlException();
```

Comment:
This is going to return AccessControlException for any error server response, even if its a server system error. 
If server is already returning AccessControlException in its response for forbidden, then can we not handle the response as other APIs do for error scenarios.

Reply:
Done